torch/csrc/Exceptions.h

```diff
@@ -69,6 +69,23 @@ struct python_error : public std::exception {
     }
   }
 
+  virtual const char* what() const noexcept override {
+    // Retrieve the error message from the value.
+    if (value != nullptr && value != Py_None) {
+      AutoGIL gil;
+      Py_XINCREF(value);
```

Comment:
Manual refcounting, blegh

Reply:
I don't see a paired decref, nor do I see a steal on value, so it looks to me like you're leaking `value`?